shared/middleware/helpers/storage_id.rb

```diff
@@ -101,26 +101,34 @@ def storage_id_for_current_user
   user_storage_id = storage_id_for_user_id(request.user_id)
   return user_storage_id unless user_storage_id.nil?
 
-  # Take ownership of cookie storage, if it exists.
-  if storage_id = storage_id_from_cookie
-    # Delete the cookie that was tracking this storage id
-    response.delete_cookie(storage_id_cookie_name)
-
-    # Only take ownership if the storage id doesn't already have an owner - it shouldn't but
-    # there is a race condition (addressed below)
-    rows_updated = user_storage_ids_table.where(id: storage_id, user_id: nil).update(user_id: request.user_id)
-    return storage_id if rows_updated > 0
-
-    # We couldn't claim the storage. The most likely cause is that another request (by this
-    # user) beat us to the punch so we'll re-check to see if we own it. Otherwise the storage
-    # id is either invalid or it belongs to another user (both addressed below)
-    return storage_id if user_storage_ids_table.where(id: storage_id, user_id: request.user_id).first
-  end
+  user_storage_id = take_storage_id_ownership_from_cookie(request.user_id)
+  return user_storage_id unless storage_id.nil?
 
   # We don't have any existing storage id we can associate with this user, so create a new one
   user_storage_ids_table.insert(user_id: request.user_id)
 end
 
+def take_storage_id_ownership_from_cookie(user_id)
+  # Take ownership of cookie storage, if it exists.
+  storage_id = storage_id_from_cookie
+  return unless storage_id
+
+  puts "Taking ownership of #{storage_id}"
```

Comment:
leftover from debugging? Or do we want this?

Reply:
leftover. thx for catching